PayloadPkg/FirmwareUpdate/GetCapsuleImage.c

```diff
@@ -269,12 +300,25 @@ LoadCapsuleImage (
   FileHandle        = NULL;
   HwPartHandle      = NULL;
   FsHandle          = NULL;
+  SwPart            = 0;
+  BootSlot = 0;
 
   Status = FindBootPartition (CapsuleInfo, &HwPartHandle);
   if (EFI_ERROR (Status)) {
     goto Done;
   }
 
+  Status = GetSwPartByName (HwPartHandle, L"misc", &SwPart);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "Unable to find misc partition, Status = %r\n", Status));
+  }
+
+  BootOption.BootFlags = BOOT_FLAGS_MISC | BOOT_FLAGS_MENDER;
+  BootOption.Image[LoadImageTypeMisc].LbaImage.SwPart = (UINT8)SwPart;
+  BootOption.Image[LoadImageTypeMisc].LbaImage.LbaAddr= 0;
+  BootOption.HwPart = CapsuleInfo->HwPart;
+  BootSlot = GetBootSlot (&BootOption, HwPartHandle);
+
```

Comment:
SBL capsule update image location is decided by the CapsuleInfo.
From the patch, it looks the expected SBL capsule image should be read from PART_A when mender OS is boot from PART_A, and read from PART_B when mender OS is boot from PART_B.
This is a new feature request, and the changes should not impact existing usage. (e.g. should not try to read boot slot info in normal capsule image loading).
I prefer adding a flag in CAPSULE_INFO_CFG_DATA to support this new feature. With this new flag, the change in this file looks like as below:
BootSlot = 0;
if (CapsuleInfo->CapsuleFlag & CAPSULE_FLAGS_MENDER !=0) {
  // update BootSlot as above
}